app/service/file_svc.py

```diff
@@ -1,19 +1,37 @@
 import os
+import base64
 
 from aiohttp import web
+from cryptography.fernet import Fernet
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+from cryptography.hazmat.backends import default_backend
 
 from app.utility.base_service import BaseService
 from app.utility.payload_encoder import xor_file
 
 
 class FileSvc(BaseService):
 
-    def __init__(self, exfil_dir):
+    def __init__(self, exfil_dir, file_encryption=True, api_key=None, crypt_salt=None):
         self.exfil_dir = exfil_dir
         self.log = self.add_service('file_svc', self)
         self.data_svc = self.get_service('data_svc')
         self.special_payloads = dict()
 
+        if file_encryption and not (api_key and crypt_salt):
+            self.log.error('File encryption requires setting api_key and crypt_salt in the config file.')
+        elif file_encryption:
+            generated_key = PBKDF2HMAC(algorithm=hashes.SHA256(),
+                                       length=32,
+                                       salt=bytes(crypt_salt, 'utf-8'),
+                                       iterations=2 ** 20,
+                                       backend=default_backend())
+            self.encryptor = Fernet(base64.urlsafe_b64encode(generated_key.derive(bytes(api_key, 'utf-8'))))
+        else:
+            self.encryptor = None
```

Comment:
move logic to a private function but keep instantiation here.

Reply:
Changes are here: 9607dabc3ee7f61b2ea0545f270f630a69731352  (CI fail was just due to some extra whitespace)